activesupport/test/file_update_checker_test.rb

```diff
@@ -10,10 +10,6 @@ def new_checker(files = [], dirs = {}, &block)
     ActiveSupport::FileUpdateChecker.new(files, dirs, &block)
   end
 
-  def wait
-    # noop
-  end
-
   def touch(files)
     sleep 1 # let's wait a bit to ensure there's a new mtime
```

Comment:
Anyone happen to know if the tests become flaky if we don't sleep for an entire second? Would `sleep 0.01` or `sleep 0.1` be long enough?

Reply:
There are actually quite a few locations that sleep for a full second that I'm seeing when I grep for `sleep` in the rails codebase. I'm wondering if they are all necessary, or if we may have ended up with some of them because `sleep 1` has been an instinctual response to address race conditions in those tests. I'm wondering if 10 or 100 milliseconds would be sufficient in avoiding test flakiness while speeding up the test suite. (Of course, it'd be even better to refactor the tests to explicitly wait just long enough for the necessary condition, but that'd certainly be a more involved change, whereas just dropping to `sleep 0.1` might be low-hanging fruit.)